loader/compat_php.c

```diff
@@ -0,0 +1,212 @@
+    }
+
+    // Fallback for PHP < 7.4
+    uint32_t nIndex;
+    uint32_t idx, i;
+    Bucket *p, *arData;
+
+    ZEND_ASSERT(!(HT_FLAGS(ht) & HASH_FLAG_PACKED));
+
+    p = (Bucket *)zend_hash_find(ht, key);
+    if (UNEXPECTED(p)) {
+        return (p == b) ? &p->val : NULL;
+    }
+
+    if (!ddloader_zstr_is_interned(php_api_no, key)) {
+        GC_ADDREF(key);
+        HT_FLAGS(ht) &= ~HASH_FLAG_STATIC_KEYS;
+    }
+
+    arData = ht->arData;
+
+    /* del from hash */
+    idx = HT_IDX_TO_HASH(b - arData);
+    nIndex = b->h | ht->nTableMask;
+    i = HT_HASH_EX(arData, nIndex);
+    if (i == idx) {
+        HT_HASH_EX(arData, nIndex) = Z_NEXT(b->val);
+    } else {
+        p = HT_HASH_TO_BUCKET_EX(arData, i);
+        while (Z_NEXT(p->val) != idx) {
+            i = Z_NEXT(p->val);
+            p = HT_HASH_TO_BUCKET_EX(arData, i);
+        }
+        Z_NEXT(p->val) = Z_NEXT(b->val);
+    }
+
+    ddloader_zend_string_release(php_api_no, b->key);
+
+    /* add to hash */
+    idx = b - arData;
+    b->key = key;
+    b->h = ZSTR_H(key);
+    nIndex = b->h | ht->nTableMask;
+    idx = HT_IDX_TO_HASH(idx);
+    i = HT_HASH_EX(arData, nIndex);
+    if (i == HT_INVALID_IDX || i < idx) {
+        Z_NEXT(b->val) = i;
+        HT_HASH_EX(arData, nIndex) = idx;
+    } else {
+        p = HT_HASH_TO_BUCKET_EX(arData, i);
+        while (Z_NEXT(p->val) != HT_INVALID_IDX && Z_NEXT(p->val) > idx) {
+            i = Z_NEXT(p->val);
+            p = HT_HASH_TO_BUCKET_EX(arData, i);
+        }
+        Z_NEXT(b->val) = Z_NEXT(p->val);
+        Z_NEXT(p->val) = idx;
+    }
+    return &b->val;
+}
+
+static void ddloader_php_70_71_zend_error_cb(int type, const char *error_filename, const uint error_lineno, const char *format, va_list args) {
+    UNUSED(type);
+    UNUSED(error_filename);
+    UNUSED(error_lineno);
+    ddloader_logv(WARN, format, args);
+}
+
+static void ddloader_php_72_73_74_zend_error_cb(int type, const char *error_filename, const uint32_t error_lineno, const char *format, va_list args) {
+    UNUSED(type);
+    UNUSED(error_filename);
+    UNUSED(error_lineno);
+    ddloader_logv(WARN, format, args);
+}
+
+static void ddloader_php_80_error_zend_error_cb(int type, const char *error_filename, const uint32_t error_lineno, zend_string *message) {
+    UNUSED(type);
+    UNUSED(error_filename);
+    UNUSED(error_lineno);
+    LOG(WARN, "Error while registering the module: %s", ZSTR_VAL(message));
+}
+
+static void ddloader_php_error_zend_error_cb(int type, zend_string *error_filename, const uint32_t error_lineno, zend_string *message) {
+    UNUSED(type);
+    UNUSED(error_filename);
+    UNUSED(error_lineno);
+    LOG(WARN, "Error while registering the module: %s)", ZSTR_VAL(message));
+}
+
+void (*old_zend_error_cb)(void);
+
+void ddloader_replace_zend_error_cb(int php_api_no) {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wincompatible-pointer-types"
+    old_zend_error_cb = zend_error_cb;
```

Comment:
```suggestion
    old_zend_error_cb = (void *)zend_error_cb;
```
You may just cast the stuff to `void` on the assignments to avoid the warning IIRC (same below)